chapters/ru/chapter9/7.mdx

```diff
@@ -0,0 +1,236 @@
+]} />
+
+В предыдущих разделах мы изучили и создали демо, используя класс `Interface`. В этом разделе мы представим наш **свеже разработанный** низкоуровневый API под названием `gradio.Blocks`.
+
+Итак, в чем разница между `Interface` и `Blocks`?
+
+- ⚡ `Interface`: высокоуровневый API, который позволяет создать полноценное демо машинного обучения, просто предоставив список входных и выходных данных.
+
+- 🧱 `Blocks`: низкоуровневый API, который позволяет вам полностью контролировать потоки данных и компоновку вашего приложения. Вы можете создавать очень сложные, многоступенчатые приложения, используя `Blocks` (как "строительные блоки").
+
+
+### Почему Blocks 🧱?[[why-blocks-]]
+
+Как мы видели в предыдущих разделах, класс `Interface` позволяет легко создавать полноценные демо машинного обучения с помощью всего нескольких строк кода. API `Interface` чрезвычайно прост в использовании, но ему не хватает гибкости, которую обеспечивает API `Blocks`. Например, вы можете захотеть:
+
+- Сгруппировать связанные демо в виде нескольких вкладок в одном веб-приложении
+- Изменить макет вашего демо, например, указать, где расположены входные и выходные компоненты
+- Многоступенчатые интерфейсы, в которых выход одной модели становится входом для следующей модели, или более гибкие потоки данных в целом
+- Изменить свойства компонента (например, выбор в выпадающем списке) или его видимость на основе пользовательского ввода
+
+Ниже мы рассмотрим все эти понятия.
+
+### Создание простого демо с помощью блоков[[creating-a-simple-demo-using-blocks]]
+
+После того как вы установили Gradio, запустите приведенный ниже код в виде сценария на Python, блокнота Jupyter или блокнота Colab.
+
+```py
+import gradio as gr
+
+
+def flip_text(x):
+    return x[::-1]
+
+
+demo = gr.Blocks()
+
+with demo:
+    gr.Markdown(
+        """
+    # Flip Text!
+    Start typing below to see the output.
+    """
+    )
+    input = gr.Textbox(placeholder="Flip this text")
+    output = gr.Textbox()
+
+    input.change(fn=flip_text, inputs=input, outputs=output)
+
+demo.launch()
+```
+
+<iframe src="https://course-demos-flip-text.hf.space" frameBorder="0" height="400" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
+В этом простом примере представлены 4 концепции, которые лежат в основе блоков:
+
+1. Блоки позволяют создавать веб-приложения, сочетающие в себе разметку, HTML, кнопки и интерактивные компоненты, просто инстанцируя объекты на Python в контексте `with gradio.Blocks`.
+<Tip>
```

Comment:
```suggestion
<Tip>

```

Reply:
It's not an intuitive thing, but there should be a new line before and after <Tip>, otherwise there are issues with rendering links inside a tip.